static/app/components/button.tsx

```diff
@@ -139,22 +142,39 @@ type StyledButtonProps = ButtonProps & {theme: Theme};
 const getFontWeight = ({priority, borderless}: StyledButtonProps) =>
   `font-weight: ${priority === 'link' || borderless ? 'inherit' : 600};`;
 
-const getBoxShadow =
-  (theme: Theme) =>
-  ({priority, borderless, disabled}: StyledButtonProps) => {
-    if (disabled || borderless || priority === 'link') {
-      return 'box-shadow: none';
-    }
+const getBoxShadow = ({
+  priority,
+  borderless,
+  translucentBorder,
+  disabled,
+  theme,
+}: StyledButtonProps) => {
+  const themeName = disabled ? 'disabled' : priority || 'default';
+  const {borderTranslucent} = theme.button[themeName];
+  const translucentBorderString = translucentBorder
+    ? `0 0 0 1px ${borderTranslucent},`
+    : '';
 
-    return `
-      box-shadow: ${theme.dropShadowLight};
+  if (disabled || borderless || priority === 'link') {
+    return 'box-shadow: none';
+  }
+
+  return `
+      box-shadow: ${translucentBorderString} ${theme.dropShadowLight};
```

Comment:
Yeah I considered it, but thought it was more flexible this way, since we can mix different box shadows types together (border + light shadow, border + heavy shadow, etc.).